lightningd/channel_control.c

```diff
@@ -77,9 +77,9 @@ static void peer_got_shutdown(struct channel *channel, const u8 *msg)
 	}
 
 	/* If we weren't already shutting down, we are now */
-	if (channel->state == CHANNELD_NORMAL)
+	if (channel->state != CHANNELD_SHUTTING_DOWN)
```

Comment:
Might a bit too permissive now? We probably also don't want to accept in state sigexchange.

Reply:
If we're in channeld (which we are), we have to be in AWAITING_LOCKIN or NORMAL or SHUTTING_DOWN.  So this is actually safe.